actionpack/test/dispatch/routing_assertions_test.rb

```diff
@@ -73,6 +86,7 @@ def test_assert_generates_with_extras
   def test_assert_recognizes
     assert_recognizes({ controller: "articles", action: "index" }, "/articles")
     assert_recognizes({ controller: "articles", action: "show", id: "1" }, "/articles/1")
+    assert_recognizes({ controller: "books", action: "index" }, "/")
```

Comment:
I think this assertion should be in its own separate test, since the two above cover basic resource routes and this one is for a fairly exotic edge case. There's some existing test coverage for engines further down.